lib/kvao/delete.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+var assert = require('assert');
+var debug = require('debug')('loopback:kvao:delete');
+var utils = require('../utils');
+
+/**
+ * Delete the key-value pair associated to the given key.
+ *
+ * @param {String} key Key to use when searching the database.
+ * @options {Object} options
+ * @callback {Function} callback
+ * @param {Error} err Error object.
+ * @param {*} result Value associated with the given key.
+ * @promise
+ *
+ * @header KVAO.prototype.delete(key[, options], cb)
+ */
+module.exports = function keyValueDelete(key, options, callback) {
+  if (callback == undefined && typeof options === 'function') {
+    callback = options;
+    options = {};
+  } else if (!options) {
+    options = {};
+  }
+
+  assert(typeof key === 'string' && key, 'key must be a non-empty string');
+
+  callback = callback || utils.createPromiseCallback();
+
+  var connector = this.getConnector();
+  if (typeof connector.delete === 'function') {
+    connector.delete(this.modelName, key, options, callback);
+  } else {
+    debug('connector does not support `delete`');
```

Comment:
I think it may be better to use the same message I proposed for "deleteAll" above, i.e. `Connector does not support deleting key/values`.